common/models/transducer/alignment.py

```diff
@@ -1,16 +1,33 @@
 from returnn.tf.util.basic import get_shape_dim, check_input_dim
+from returnn.tf.util.data import Data
+from returnn.tf.layers.base import LayerBase
+from tensorflow import Tensor
+from typing import List, Union, Protocol
 
 
-def rna_alignment(source, **kwargs):
+class SourceType(Protocol):
```

Comment:
This is how I end up doing. We can maybe add a file called `common_types.py` somewhere to save complicated type definitions such as this.